### DISCUSSION TITLE
How to show correct data in different map (4326 &3857)？

### DISCUSSION THREAD

**Discussion by Emelia-L** [visgl/deck.gl#5539] [Q&A]
I want to show a layer on the map in Deck. 
When I use mapbox,  the data can show correctly as below.
![image](https://user-images.githubusercontent.com/41425832/109284684-16836900-785b-11eb-9e64-b8efdabd2506.png)
But when I want to show the data in GeojsonLayer on 4326 map in Deck,  because of web-mercator projection, the data is not consist with the 4326 map as below.
![image](https://user-images.githubusercontent.com/41425832/109284867-4d597f00-785b-11eb-95dd-6d2af79db653.png)
I wonder if there is some methods to show data correctly in Deck layers with 4326 map?

**Comment by Pessimistress** [comment]
I'm not sure I understand your question. What is the input coordinate system of your data, and what is the output projection that you are trying to use? Can you add a code snippet alongside your screenshots?

**Reply by Emelia-L** [reply]
My input data is Geojson data with type "MultiPolygon", and I use GeojsonLayer to show it. 
When I use mapbox, Geojsonlayer is consist with the map:
![image](https://user-images.githubusercontent.com/41425832/109303395-b4842d00-7875-11eb-9ecb-b1f2980afc15.png)

But I don't want to use mapbox, so I add the 4326 settings of _mercator_coordinate.js_ in mapbox-gl  as shown below:
![image](https://user-images.githubusercontent.com/41425832/109301155-82250080-7872-11eb-8893-8d01d14bb659.png)
![image](https://user-images.githubusercontent.com/41425832/109301273-b39dcc00-7872-11eb-8dc7-a4c819dca518.png)
And then, I also change the tiles address, so I can replace mapbox with a 4326 map.  
Then, I got the result:
![image](https://user-images.githubusercontent.com/41425832/109302939-f82a6700-7874-11eb-99b7-4547c6cf2a3a.png)
It shows that the Geojsonlayer on my screen is not consist with the 4326 map. I think it is because  mapview in Deck is based on web Mercator. I want to show correct GeojsonLayer  on 4326 map, what should I do?

**Reply by Pessimistress** [reply]
same as #5513

`MapView` uses WebMercator projection. If you don't want to use WebMercator, then use `OrthographicView` instead. Here's an example by @tech-meppem: https://gist.github.com/tech-meppem/5a40e00b04220e924177791b4982be0f

You can use [modelMatrix](https://deck.gl/docs/api-reference/core/layer#modelmatrix) to scale/translate the data to match the base map.
